src/Impl/GuzzleHttpRequester.php

```diff
@@ -44,27 +59,37 @@ public function executePost(
      * @return MessageStatus[]
      * @throws StreamxClientException
      */
-    private function handleResponse(ResponseInterface $response): array
+    private function handleIngestionResponse(ResponseInterface $response): array
     {
         $statusCode = $response->getStatusCode();
 
-        if ($statusCode == 202) {
-            return $this->parseMessageStatuses($response);
-        }
-
-        if ($statusCode == 401) {
-            throw new StreamxClientException('Authentication failed. Make sure that the given token is valid.');
+        switch ($statusCode) {
+            case 202:
+                return $this->parseMessageStatuses($response);
+            case 401:
+                throw $this->createAuthenticationException();
+            case in_array($statusCode, [400, 403, 500]):
+                $failureResponse = $this->parseFailureResponse($response);
+                throw $this->streamxClientExceptionFrom($failureResponse);
+            default:
+                throw $this->createGenericCommunicationException($response);
         }
+    }
 
-        if (in_array($statusCode, [400, 403, 500])) {
-            $failureResponse = $this->parseFailureResponse($response);
-            throw $this->streamxClientExceptionFrom($failureResponse);
+    /**
+     * @throws StreamxClientException
+     */
+    private function handleSchemaResponse(ResponseInterface $response): string
+    {
+        $statusCode = $response->getStatusCode();
+        switch ($statusCode) {
+            case 200:
+                return (string)$response->getBody();
+            case 401:
+                throw $this->createAuthenticationException();
+            default:
+                throw $this->createGenericCommunicationException($response);
```

Comment:
Schema endpoint in StreamX declares it returns also some other response codes:
```java
      @APIResponse(responseCode = "200", description = "Channels refreshed successfully"),
      @APIResponse(responseCode = "400", description = "Channel unsupported"),
      @APIResponse(responseCode = "403", description = "Channel forbidden"),
      @APIResponse(responseCode = "500", description = "Error while getting channel")
```
But there should not be a need to handle each separately, `createGenericCommunicationException` _should_ print the status code and error message.

For example when creating publisher with a channel that doesn't exist:

> Streamx\Clients\Ingestion\Exceptions\StreamxClientException : Communication error. Response status: 400. Message: Bad Request